src/main/java/hudson/plugins/mercurial/HgExe.java

```diff
@@ -391,13 +404,19 @@ public String config(FilePath repository, String name) throws IOException, Inter
     /**
      * Runs the command and captures the output.
      */
+    @NonNull
     public String popen(FilePath repository, TaskListener listener, boolean useTimeout, ArgumentListBuilder args)
             throws IOException, InterruptedException {
         args = seed(false).add(args.toCommandArray());
 
         ByteArrayOutputStream data = new ByteArrayOutputStream();
         if (joinWithPossibleTimeout(launch(args).pwd(repository).stdout(data), useTimeout, listener) == 0) {
-            return data.toString();
+            try {
+                //TODO: consider using another charset
+                return data.toString(Charset.defaultCharset().name());
+            } catch (UnsupportedCharsetException ex) { // Should never happen
+                throw new IOException("Cannot perform a conversion using the default charset", ex);
+            }
```

Comment:
I disagree with @jglick on using `@SuppressFBWarnings` vs explicitly using `Charset.defaultCharset`.
The supression will affect the whole method (unless you are using JDK8 and can apply it on a more fine grained scope) - and blanket suppressions normally lead to issues as one call may need to use a different encoding...

It also means the code is explicit where you read it rather than needing to scroll up to see the method declaration.